contrib/epee/src/net_ssl.cpp

```diff
@@ -494,6 +495,7 @@ bool ssl_options_t::handshake(boost::asio::ssl::stream<boost::asio::ip::tcp::soc
         if (support != ssl_support_t::e_ssl_support_autodetect)
         {
           MERROR("SSL certificate is not in the allowed list, connection droppped");
+          failed = true;
           return false;
```

Comment:
ec does not get set. I double checked that. If it does for you, I have openssl 1.1.0i here.

Reply:
What version of boost?

Reply:
1.70.0, though that is a beta IIRC.

Reply:
An error is printed on the server with the connection being dropped yet the client returns no error code. I wonder who is responsible for this mess, seems like something Boost. Although I peeked, and OpenSSL re-wrote a bunch of relevant code between 1.0 and 1.1 too.

Reply:
This is a bug that exists only in the beta/rc version of boost 1.70. The conditional on line 514 can never be reached by any official Boost releases. Whether this check should be left as a pessimism ... ?

Anyway, good news because I'm not obligated to patch upstream anymore! Yay beta testers. Also, probably want to update your local boost to 1.70 final release.

Reply:
OK, so extreme bad luck then. Thanks for hunting it. If it only happens with that particular version, I don't mind whether to keep or remove it.